libbeat/cfgfile/reload.go

```diff
@@ -15,3 +31,163 @@ type ReloadConfig struct {
 	Period  time.Duration `config:"period"`
 	Enabled bool          `config:"enabled"`
 }
+
+type RunnerFactory interface {
+	Create(*common.Config) (Runner, error)
+}
```

Comment:
having just one method how about:

```
type RunnerFactory func(*common.Config) (Runner, error)
```

This is already a very common pattern in beats.